src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsService.kt

```diff
@@ -438,17 +439,19 @@ class AdjudicationsService(
 
   private inline fun <reified T> String.fromJson(): T =
     objectMapper.readValue(this)
-}
 
-private fun PunishmentDto.toNomisAward() = CreateHearingResultAwardRequest(
-  sanctionType = this.type.toNomisSanctionType(),
-  sanctionStatus = this.toNomisSanctionStatus(),
-  effectiveDate = this.schedule.startDate ?: this.schedule.suspendedUntil ?: LocalDate.now(),
-  sanctionDays = this.schedule.days,
-  compensationAmount = this.damagesOwedAmount?.toBigDecimal() ?: stoppagePercentage?.toBigDecimal(),
-  commentText = this.toComment(),
-  // TODO consecutive adjudication
-)
+  private suspend fun PunishmentDto.toNomisAward() = CreateHearingResultAwardRequest(
+    sanctionType = this.type.toNomisSanctionType(),
+    sanctionStatus = this.toNomisSanctionStatus(),
+    effectiveDate = this.schedule.startDate ?: this.schedule.suspendedUntil ?: LocalDate.now(),
+    sanctionDays = this.schedule.days,
+    compensationAmount = this.damagesOwedAmount?.toBigDecimal() ?: stoppagePercentage?.toBigDecimal(),
+    commentText = this.toComment(),
+    consecutiveCharge = this.consecutiveChargeNumber?.let { chargeNumber ->
+      adjudicationMappingService.getMappingGivenChargeNumber(chargeNumber).let { AdjudicationChargeId(it.adjudicationNumber, it.chargeSequence) }
```

Comment:
hmm, not sure - since the first it saying:
transform this if it exists
the second is saying:
Now I know i'm going to transform - get the data the convert

So I see them as nested - even though chaining would work perfectly, but you would have to change the `.let` to `?.let`

I think this is maybe just a personal style thing - and next week I'll do the opposite. 
Good point though